src/pykeen/pipeline/api.py

```diff
@@ -999,6 +1011,31 @@ def pipeline(  # noqa: C901
     else:
         mapped_triples = validation.mapped_triples
 
+    # Determine whether the validation triples should also be filtered while performing test evaluation
+    if (
+        evaluator_instance.filtered
+        and use_testing_data
+        and filter_validation_when_testing
+        and validation is not None
+    ):
+        logging.info(
+            "When evaluating the test dataset, validation triples are added to the set of known positive triples"
+            " which are filtered out when performing filtered evaluation following the approach described by"
+            " (Bordes et al., 2013).",
+        )
+        additional_filter_triples = validation.mapped_triples
+
+        if 'additional_filter_triples' not in evaluation_kwargs:
+            evaluation_kwargs['additional_filter_triples'] = additional_filter_triples
+        else:
+            logger.info(
+                'there were already `additional_filter_triples` in the evaluation arguments,'
+                ' so the custom ones will be concatenated',
+            )
+            evaluation_kwargs['additional_filter_triples'] = torch.cat(
+                [evaluation_kwargs['additional_filter_triples'], additional_filter_triples], dim=0
```

Comment:
should there be a unique filer here? or does this not matter?

Reply:
We have to check if it really matters. Right now we make sure that we do not create duplicates by concatenating the same dataset twice, so it would only gain something when the dataset itself has a lot of duplicates.

Reply:
Okay. This isn't a blocker for the PR, but I will leave a todo for later